### ISSUE TITLE
Config for pretraining Wav2Vec2

### ISSUE THREAD

**Issue by thechuong98** [kehanlu/Mandarin-Wav2Vec2#1]
Hi @kehanlu,

I'm pretraining a Wav2Vec2 base model on Vietnamese internal dataset (around 2.5k hours) using 8 32GB V100 GPU. I'm also using the base model config as on fairseq with an increase of effective batch size (2.560.000 compared to 1.400.000 default) and I set the init learning rate as 0.005 (it's 0.0005 in the default config). However the training is very unstable 
<img width="329" alt="image" src="https://user-images.githubusercontent.com/60946216/170917845-20ccc5bc-16f5-4bd2-b363-cdb5170439c0.png">
Could you share about your config or your training best practice?
Thank you!

**Comment by thechuong98** [comment]
Also, if I train the base LibriSpeech config with 8 V100 GPU , the estimated total training steps for 10 days would be 50k, but in paper it's 400k. I saw you reached 400k steps as in fairseq (used 64 V100 GPU). Can you share the log of your training procedure? 
Thank you so much!

**Comment by kehanlu** [comment]
Hi @thechuong98,

1. We used `max_tokens: 1400000`, `lr: [0.0005]` same as the base model config from Fairseq. We did concat 6 wav files to 1 single file, because wavfiles in Librispeech are longer than the dataset we used(AISHELL-2). Our config is actually from the authors of previous works. It is almost the same as Fairseq. We haven't try other configs yet.
I think using smaller learning rate will stabilize the training process(I read from other issue before, but I can't find it now). 

2. We use 8 V100 GPUs and accumulate 8 updates(`update_freq: 8`) to simulate 64 GPUs (according to [here](https://github.com/facebookresearch/fairseq/blob/main/examples/wav2vec/README.md#train-a-wav2vec-20-base-model)). Did you configure it correctly?
I can't reach to the server right now. I will upload the logs as soon as I can.

Thank you

**Comment by thechuong98** [comment]
Thank you for your detailed reply! I also accumulated 8 updates for each GPU but it took 1 hour for each 200 num_updates, I havent figured out why yet :D 

Here is the cli I used for pretraining, is that quite the same as yours?

CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 fairseq-hydra-train     task.data=unlabeled_data    distributed_training.distributed_world_size=8    +optimization.update_freq='[8]'    --config-dir fairseq/examples/wav2vec/config/pretraining/     --config-name wav2vec2_base_librispeech

**Comment by kehanlu** [comment]
Can you check that all additional configs are properly overwrite by CLI in your training log file? Since I have this kind of issue, then I modified the config file directly.

BTW, we use the commit [`313ff0581561c7725ea9430321d6af2901573dfb` ](https://github.com/facebookresearch/fairseq/commit/313ff0581561c7725ea9430321d6af2901573dfb) version, which was installed alongside Espnet(tools/installer/install_fairseq.sh).

**Comment by kehanlu** [comment]
I have uploaded part of the trianing log. You can find the config and some other information you might need in the file.

**Comment by thechuong98** [comment]
That's all I need. Thank you :D
